presto-kafka/src/main/java/io/prestosql/plugin/kafka/encoder/raw/RawRowEncoder.java

```diff
@@ -0,0 +1,345 @@
+        for (EncoderColumnHandle handle : this.columnHandles) {
+            columnMappingsBuilder.add(new ColumnMapping(handle));
+        }
+        this.columnMappings = columnMappingsBuilder.build();
+
+        // make sure the actual/expected lengths match
+        for (ColumnMapping mapping : this.columnMappings) {
+            if (mapping.getLength() != mapping.getFieldType().getSize() && !isVarcharType(mapping.getType())) {
+                throw new IndexOutOfBoundsException(format(
+                        "Mapping length '%s' is not equal to expected length '%s' for column '%s'",
+                        mapping.getLength(),
+                        mapping.getFieldType().getSize(),
+                        mapping.getName()));
+            }
+        }
+
+        // check that column mappings don't overlap and that there are no gaps
+        int position = 0;
+        for (ColumnMapping mapping : this.columnMappings) {
+            checkArgument(mapping.getStart() == position, format(
+                    "Start mapping '%s' for column '%s' does not equal expected mapping '%s'",
+                    mapping.getStart(),
+                    mapping.getName(),
+                    position));
+            checkArgument(mapping.getEnd() > mapping.getStart(), format(
+                    "End mapping '%s' for column '%s' is less than or equal to start '%s'",
+                    mapping.getEnd(),
+                    mapping.getName(),
+                    mapping.getStart()));
+            position += mapping.getLength();
+        }
+
+        this.buffer = ByteBuffer.allocate(position);
+    }
+
+    private static class ColumnMapping
+    {
+        private final String name;
+        private final Type type;
+        private final int start;
+        private final int end;
+        private final FieldType fieldType;
+
+        public ColumnMapping(EncoderColumnHandle columnHandle)
+        {
+            this.name = columnHandle.getName();
+            this.type = columnHandle.getType();
+            this.fieldType = parseFieldType(columnHandle.getDataFormat(), this.name);
+            checkFieldType(this.name, this.type, this.fieldType);
+            Optional<String> mapping = Optional.ofNullable(columnHandle.getMapping());
+
+            if (mapping.isPresent()) {
+                Matcher mappingMatcher = MAPPING_PATTERN.matcher(mapping.get());
+                if (!mappingMatcher.matches()) {
+                    throw new IllegalArgumentException(format("Invalid mapping for column '%s'", this.name));
+                }
+                else if (mappingMatcher.group(2) != null) {
+                    this.start = parseOffset(mappingMatcher.group(1), "start", this.name);
+                    try {
+                        this.end = parseOffset(mappingMatcher.group(2), "end", this.name);
+                    }
+                    catch (NumberFormatException e) {
```

Comment:
It doesn't look like the method throws `NumberFormatException`.  I also don't get the exception message.  But it doesn't look like you need to catch here.

Reply:
oh right, I moved the error handling to the method and forgot to remove it here